feature/onboarding/src/commonMain/kotlin/com/nexters/emotia/feature/onboarding/OnBoardingScreen.kt

```diff
@@ -114,104 +131,333 @@ private fun LoginContent(
+                    .align(Alignment.Center)
+                    .offset(y = firstImageOffsetY),
+                contentScale = ContentScale.Crop
+            )
+        }
+
+        // Step 2: second 이미지 (하단)
+        AnimatedVisibility(
+            visible = currentStep == 1, // Step 2에서만 보임
+            enter = slideInVertically(animationSpecIntOffset) { it / 2 } + fadeIn(animationSpecFloat),
+            exit = fadeOut(animationSpecFloat)
+        ) {
+            Image(
+                painter = painterResource(Res.drawable.second),
+                contentDescription = "Second screen",
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(screenHeight / 2)
+                    .align(Alignment.BottomCenter),
+                contentScale = ContentScale.Crop
+            )
+        }
+
+// Step 3: third(하단) + second(상단) 스택
+        AnimatedVisibility(
+            visible = currentStep == 2, // Step 3에서만 보임
+            enter = fadeIn(animationSpecFloat),
+            exit = fadeOut(animationSpecFloat)
+        ) {
+            Column(modifier = Modifier.fillMaxSize()) {
+                // 상단 절반 - second 이미지 (투명도 50%)
+                Image(
+                    painter = painterResource(Res.drawable.second),
+                    contentDescription = "Second screen (top half)",
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .weight(1f), // 절반 차지
+                    contentScale = ContentScale.Crop,
+                    alpha = 0.5f
+                )
+
+                // 하단 절반 - third 이미지
+                Image(
+                    painter = painterResource(Res.drawable.third),
+                    contentDescription = "Third screen (bottom half)",
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .weight(1f), // 절반 차지
+                    contentScale = ContentScale.Crop
+                )
+            }
+        }
+
+
+        // Step 4: four 이미지로 전환
+        AnimatedVisibility(
+            visible = currentStep == 3,
+            enter = fadeIn(animationSpecFloat),
+            exit = fadeOut(animationSpecFloat)
+        ) {
+            Image(
+                painter = painterResource(Res.drawable.four),
+                contentDescription = "Fourth screen",
+                modifier = Modifier.fillMaxSize(),
+                contentScale = ContentScale.Crop
+            )
+        }
+
+        // Step 4
+        AnimatedVisibility(
+            visible = currentStep == 4,
+            enter = fadeIn(animationSpecFloat),
+            exit = fadeOut(animationSpecFloat)
+        ) {
+            Image(
+                painter = painterResource(Res.drawable.five),
+                contentDescription = "Fifth screen",
+                modifier = Modifier.fillMaxSize(),
+                contentScale = ContentScale.Crop
+            )
+        }
```

Comment:
```suggestion
    @Composable
fun StepImage(
    currentStep: Int,
    targetStep: Int,
    drawableRes: DrawableResource,
    description: String,
    animationSpec: FiniteAnimationSpec<Float>
) {
    AnimatedVisibility( 
    // 이하 코드 )
```

스텝 12까지 중복된 코드가 많아요 이런식으로 묶어주세요 !